src/Chapter3.hs

```diff
@@ -971,6 +1097,30 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data Weekday
+   = Mon
+   | Tue
+   | Wed
+   | Thu
+   | Fri
+   | Sat
+   | Sun
+   deriving (Show, Eq, Enum, Bounded)
+
+isWeekend :: Weekday -> Bool
+isWeekend wd = case wd of
+    Sat -> True
+    Sun -> True
+    _ -> False
+
+nextDay :: Weekday -> Weekday
+nextDay wd
+    | wd == maxBound = minBound
+    | otherwise = succ wd
+
+daysToParty :: Weekday -> Int
+daysToParty wd = abs (fromEnum wd - fromEnum Fri)
```

Comment:
This would show the shortest distance to `Friday`, however, time doesn't work like this.
For example. if it is `Saturday`, it would tell you that it is 1 day to `Friday`, however, it is 6 days to go 🥳